tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,343 @@
+            need_restore_state = True
+
+        # Currently we select 'rsyslogd' as non-critical processes for testing based on
+        # the assumption that every container has an 'rsyslogd' process running and it is not
+        # considered to be a critical process
+        program_status, program_pid = get_program_info(duthost, container_name, "rsyslogd")
+        verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd",
+                                                        program_status, program_pid)
+
+        critical_group_list, critical_process_list = get_critical_group_and_process_list(duthost, container_name)
+        for critical_process in critical_process_list:
+            program_status, program_pid = get_program_info(duthost, container_name, critical_process)
+            verify_autorestart_with_critical_process(duthost, container_name, critical_process,
+                                                     program_status, program_pid)
+            # Sleep 20 seconds in order to let the processes come into live after container is restarted.
+            # We will uncomment the following line once the "extended" mode is added
+            # time.sleep(20)
+            # We are currently only testing one critical process, that is why we use 'break'. Once
+            # we add the "extended" mode, we will remove this statement
+            break
+
+        for critical_group in critical_group_list:
+            group_program_info = get_group_program_info(duthost, container_name, critical_group)
+            for program_name in group_program_info:
+                verify_autorestart_with_critical_process(duthost, container_name, program_name,
+                                                         group_program_info[program_name][0],
+                                                         group_program_info[program_name][1])
+                # We are currently only testing one critical program for each critical group, which is
+                # why we use 'break' statement. Once we add the "extended" mode, we will remove this
+                # statement
+                break
+
+        # After these two containers are restarted, we need wait to give their dependent containers
+        # a chance to restart
+        if container_name in ["syncd", "swss"]:
+            logger.info("Sleep 20 seconds after testing the container '{}'...".format(container_name))
+            time.sleep(20)
+
+        if need_restore_state:
+            logger.info("Restore auto-restart state of container '{}' to be '{}'"
+                        .format(container_name, container_autorestart_states[container_name]))
+            duthost.shell("config container feature autorestart {} {}"
+                          .format(container_name, container_autorestart_states[container_name]))
+
+        logger.info("End of testing the container '{}'".format(container_name))
+
```

Comment:
Delete this empty line

Reply:
I will delete it. The reason I added this empty line is when I checked this script against PEP8, it shows a warning message to indicate an empty line should be added at the end of file.

Reply:
That's unfortunately a limitation of the PEP8 website. You paste your code in to the form rather than providing it with an actual file to scan. So the file may have a newline at the end, but when pasted, it isn't there.

Reply:
Got it. Thanks, Joe! I will use autopep8 to check the file rather than copy and paste.